src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIAppWindowCodeFixer.cs

```diff
@@ -54,30 +55,42 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             context.RegisterCodeFix(
                 CodeAction.Create(
                     diagnostic.Id,
-                    c => diagnostic.Id == WinUIAppWindowAnalyzer.DiagnosticIdAppWindowNamespace ? FixAppWindowNamespace(context.Document, root, declaration, apiId!, c)
-                    : diagnostic.Id == WinUIAppWindowAnalyzer.DiagnosticIdAppWindowVarType ? FixAppWindowVarType(context.Document, declaration, c)
-                    : FixAppWindowMember(context.Document, declaration.AncestorsAndSelf().OfType<InvocationExpressionSyntax>().First(), apiId!, varName!, c),
+                    c => diagnostic.Id == WinUIAppWindowAnalyzer.DiagnosticIdAppWindowType ? FixTypeName(context.Document, root, declaration, apiId!, c)
+                    : diagnostic.Id == WinUIAppWindowAnalyzer.DiagnosticIdAppWindowVarType ? FixVarType(context.Document, declaration, c)
+                    : FixMember(context.Document, (MemberAccessExpressionSyntax)declaration, apiId!, varName!, c),
                     diagnostic.Id + apiId ?? string.Empty),
                 context.Diagnostics);
         }
 
-        private static async Task<Document> FixAppWindowNamespace(Document document, SyntaxNode oldRoot, SyntaxNode invocationExpressionSyntax, string apiId, CancellationToken cancellationToken)
+        private static async Task<Document> FixTypeName(Document document, SyntaxNode oldRoot, SyntaxNode oldNode, string apiId, CancellationToken cancellationToken)
         {
-            var comment = await CSharpSyntaxTree.ParseText(@$"
+            await Task.Yield();
+            var comment = SyntaxFactory.Comment(@$"
                 /*
-                   TODO: Use Microsoft.UI.Windowing.AppWindow for window Management instead of ApplicationView/CoreWindow or Microsoft.UI.Windowing.AppWindow APIs
+                   TODO {WinUIAppWindowAnalyzer.DiagnosticIdAppWindowType} Use Microsoft.UI.Windowing.AppWindow for window Management instead of ApplicationView/CoreWindow or Microsoft.UI.Windowing.AppWindow APIs
                    Read: https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/guides/windowing
                 */
-                ", cancellationToken: cancellationToken).GetRootAsync(cancellationToken).ConfigureAwait(false);
+                ");
 
-            var (newNamespace, newName) = WinUIAppWindowAnalyzer.TypeConversions[apiId].Value;
-            var node = (await CSharpSyntaxTree.ParseText($"{newNamespace}.{newName} x;", cancellationToken: cancellationToken)
-                .GetRootAsync(cancellationToken).ConfigureAwait(false))
-                .DescendantNodesAndSelf().OfType<QualifiedNameSyntax>().First();
-            return document.WithSyntaxRoot(oldRoot.ReplaceNode(invocationExpressionSyntax, node.WithLeadingTrivia(comment.GetLeadingTrivia())));
+            var apiConversion = WinUIAppWindowAnalyzer.TypeApiConversions[apiId];
+            if (apiConversion.ToApi is not TypeDescription)
```

Comment:
any chance `apiConversion` is `null` here?

Reply:
This cannot be null because we send the `apiId` from the Analyzer from the fixed pool of valid ids. If this fails, we know we have a bug in our logic.